azkaban-common/src/main/java/azkaban/utils/YarnUtils.java

```diff
@@ -107,17 +112,32 @@ public static List<ApplicationReport> getAllAliveAppReportsByExecIDs(final YarnC
 
 
   /**
-   * Uses YarnClient to kill the jobs one by one
+   * Uses YarnClient to kill the jobs one by one, each kill has a timeout
    */
   public static void killAllAppsOnCluster(YarnClient yarnClient, Set<String> applicationIDs,
       Logger log) {
     log.info(String.format("Killing applications: %s", applicationIDs));
+    ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
I am curious to know the reasoning behind single thread executor. would it be helpful if we use a multiple threads?